src/librustc_typeck/check/wfcheck.rs

```diff
@@ -176,9 +176,75 @@ pub fn check_trait_item(tcx: TyCtxt<'_>, def_id: DefId) {
         hir::TraitItemKind::Method(ref sig, _) => Some(sig),
         _ => None,
     };
+    check_object_unsafe_self_trait_by_name(tcx, &trait_item);
     check_associated_item(tcx, trait_item.hir_id, trait_item.span, method_sig);
 }
 
+fn could_be_self(trait_name: Ident, ty: &hir::Ty<'_>) -> bool {
+    match ty.kind {
+        hir::TyKind::TraitObject([trait_ref], ..) => {
+            let mut p = trait_ref.trait_ref.path.segments.iter().map(|s| s.ident);
+            match (p.next(), p.next()) {
```

Comment:
Why not use `trait_ref.trait_ref.path.segments.last()`?

Reply:
Because I wanted to verify that there's a single element. It's the `Iterator` equivalent to `match slice { [ident] => {}, _ => {}}`.